File: 27. Remove Element/remove_element.java
//using System;

class remove_element{
	public static int remove_by_shifting (int element, int [] list){
		int count = 0;
		int len_list = list.length;
		
		for (int i = (len_list - 1); i >= 0; i--){
			if (list[i] == element){
				count ++;
				int holder = list[i];
				
				for (int j = i; j < (len_list - 1); j++){
					list[j] = list[j + 1];
				}
				
				list[len_list - 1] = holder;
			}
		}
		
		return (count);
		
	}
	
	public static void main (String [] args){
		int [] nums = {0,1,2,2,3,0,4,2};
		int len_nums = nums.length;
		int vals = 2;
		
		int rem = remove_by_shifting(vals, nums);
		
		System.out.println("This is the handy work of Java\n\n");
		System.out.println(rem);
		System.out.println("\n\n\n");
		
		System.out.print("[");
		for (int i = 0; i < len_nums; i++){
			System.out.print(nums[i]);
			
			if (i == (len_nums - 1)){
				System.out.println("]");
			} else {
				System.out.print(", ");
			}
		}
		
		
		System.out.print("\n\n\n[");
		for (int i = 0; i < (len_nums - rem); i++){
			System.out.print(nums[i]);
			
			if (i == ((len_nums - rem) - 1)){
				System.out.println("]");
			} else {
				System.out.print(", ");
			}
		}
		System.out.println("\n\n\n");
	}
}

File: 27. Remove Element/remove_element.rs
fn remove_by_shifting (element: isize, list: &mut [isize]) -> isize{
	let mut count: isize = 0;
	let len_list: usize = list.len();
	
	for i in (0..len_list).rev(){
		if (list[i] == element){
			count += 1;
			let holder: isize = list[i];
			
			for j in i..(len_list - 1){
				list[j] = list[j + 1];
			}
			
			list[len_list - 1] = holder;
		}
	}
	
	return (count);
}

fn main (){
	let mut nums = [0,1,2,2,3,0,4,2];
	let len_nums: usize = nums.len();
	
	let vals: isize = 2;
	
	let rem: isize = remove_by_shifting(vals, &mut nums);
	
	println!("This is the handy work of Rust\n\n");
	
	println!("{}\n\n\n", rem);
	
	println!("{:?}\n\n\n", nums);
	
	
	print!("[");
	
	for i in 0..(len_nums - rem as usize){
		print!("{}", nums[i]);
		
		if (i == ((len_nums - rem as usize) - 1)){
			println!("]");
		} else {
			print!(", ");
		}
	}
	
}
